### PR TITLE
Jupyterlab 2.0

### LINKED ISSUE #438: Merge JupyterLab 2.0 branch to master
Now that Git has released a 0.20.0 release candidate which supports JupyterLab 2.x we can start rebasing and merge the JupyterLab 2.0 branch to Elyra master.

### PR DESCRIPTION
**Pull request by lresende**
Bring support for JupyterLab 2.0 to Elyra.

This currently has the latest contents of Master branch, updates required
to support JupyterLab 2.0 as well as migrations to dependencies versions
that support JupyterLab 2.0 (e.g. TOC 3.x and Git 0.20.x).

Before you can test this, you need to `clean your 1.x dev env) and install
JupyterLab 2.x.

The branch currently has two separate commits to indicate the recent 
changes related to rebase/new git extension release, but will be all
squashed before merging. 

Fixes #438 
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,10 +38,10 @@
     "lerna": "^3.16.4",
     "lint-staged": "^9.5.0",
     "prettier": "^1.19.1",
-    "rimraf": "^2.6.3",
+    "rimraf": "~3.0.0",
```
Comment (karlaspuldaro) [inline package.json:41]: should this be "~3.0.2" to match rimraf version in packages/application/package.json ?

**Comment by ajbozarth** [comment]
Going to make some notes here for things I find that I plan to fix:
- Pipeline icon no longer working

File: package.json (hunk 1 above)
Comment (ajbozarth) [reply package.json:41]: actually its the opposite, application is the one that's different than the rest, I'll catch this in my upcoming commit

Comment (lresende) [reply package.json:41]: yes, I updated all to use `"rimraf": "~3.0.0"`

**Comment by ajbozarth** [comment]
I've pushed some updates to address issues I've found. The commit addresses:

- Pipeline icon missing everywhere but the editor tab
- Arrows missing in code snippets
- Run python button not working properly (regression on the lock during rebase)
- Uncommented git install in make file

I opened #443 as a follow up issue to unblock this one.

**Comment by lresende** [comment]
@ajbozarth does this mean you are good with what we have now?

**Comment by vabarbosa** [comment]
also the `PipelineEditorWidget.tsx` has an [import from `@jupyterlab/ui-components`](https://github.com/elyra-ai/elyra/blob/jupyterlab-2.0/packages/pipeline-editor/src/PipelineEditorWidget.tsx#L38), however, it does not explicitly include `@jupyterlab/ui-components` in its [`package.json` `dependencies` field](https://github.com/elyra-ai/elyra/blob/jupyterlab-2.0/packages/pipeline-editor/package.json#L33-L58)

**Pull request merged**